src/main/kotlin/dev/arunvelsriram/ktlint/rulesets/junit/MaximumIgnoredTestRule.kt

```diff
@@ -0,0 +1,35 @@
+package dev.arunvelsriram.ktlint.rulesets.junit
+
+import com.pinterest.ktlint.core.Rule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.stubs.elements.KtStubElementTypes
+
+class MaximumIgnoredTestRule : Rule("maximum-ignored-test") {
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        if (ignoredTestsCount > IGNORED_TESTS_THRESHOLD) return
+        if (node.elementType == KtStubElementTypes.FUNCTION) {
+            val func = node.psi as KtFunction
+            func.annotationEntries.find {
+                it.text == IGNORE_ANNOTATION
```

Comment:
should we check if the function is annotated with `@Test`?

Reply:
I think it's okay because there can be ignored tests without `@Test` annotation.